server/jwt.go

```diff
@@ -105,6 +105,24 @@ func validateTrustedOperators(o *Options) error {
 			return fmt.Errorf("system_account in config and operator JWT must be identical")
 		}
 	}
+	srvMajor, srvMinor, srvUpdate, _ := jwt.ParseServerVersion(strings.Split(VERSION, "-")[0])
+	for _, opc := range o.TrustedOperators {
+		if major, minor, update, err := jwt.ParseServerVersion(opc.AssertServerVersion); err != nil {
+			return fmt.Errorf("operator %s expects version %s got error instead: %s",
+				opc.Subject, opc.AssertServerVersion, err)
+		} else if major > srvMajor {
+			return fmt.Errorf("operator %s expected major version %d > server major version %d",
+				opc.Subject, major, srvMajor)
+		} else if srvMajor > major {
+		} else if minor > srvMinor {
```

Comment:
@kozlovic I would have imagined that you'd specify x.0.0 if you only care for the major version. You don't necessarily have to  specify all the way down.  Being able to specify all the way down to a bug fix would be an escape hatch.

I can make this an integer for major version only. In that case I'd change the jwt accordingly... 
@aricart Comments?

Reply:
Understood, but will come a day when someone will update the version not knowing the implications in the server, and suddenly a bug fix in the jwt would prevent a server from using it. I am not saying that we don't use simver, although I think that @aricart was more of the opinion of using a simple integer, but if we do use simver, I think we should "fail" only for the major.

Reply:
I think it should be a simple integer that says the max 'model' version. This is a simple one or nothing because any number greater than the supported/mapped number is not compatible.

Reply:
ok, change to jwt coming, stay tuned

Reply:
Created the [PR for jwt](https://github.com/nats-io/jwt/pull/81) and have the change in this one ready

Reply:
I retract that comment - we need to do semver as in the previous PR.

Reply:
Ok, after our discussion offline, I understand better the goal and this looks good to me.